packages/cyberstorm/src/components/CustomSelect.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { ReactNode } from "react";
+import styles from "./componentStyles/CustomSelect.module.css";
+import { Button } from "./Button";
+import { CustomSelectItem, SelectItemProps } from "./CustomSelectItem";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as Select from "@radix-ui/react-select";
+import { SelectItem } from "@radix-ui/react-select";
+
+export type SelectOption = {
+  value: string;
+  content: typeof CustomSelectItem;
+  displayValue?: string;
+};
+
+type SelectProps = {
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultOpen?: boolean;
+  icon?: ReactNode;
+  options?: SelectOption[];
+  setValue: CallableFunction;
```

Comment:
I think it would make more sense name the prop as `onChange`, since that makes more sense to the developer using the component. Unless there's already an establlished practice in Cyberstorm to name the props as setFoo.

I would prefer `Dispatch<SetStateAction<string>>;` (or `string|number` if numbers are supported) since it's more specific and conveys the intent more clearly to the developer. Unless there's some specific use case that demands using the lenient `CallableFunction`.

Reply:
👍